### PR TITLE
Fixes #RHIROS-988 - Add recommender service in ros-ocp

### PR DESCRIPTION
**Pull request by patilsuraj767**

### CODE REVIEW

File: internal/logging/logging.go [hunk 1]
```diff
@@ -28,5 +29,10 @@ func InitLogger() {
 }
 
 func GetLogger() *logrus.Logger {
```
Comment (upadhyeammit) [inline internal/logging/logging.go:31]: I just felt we can combine `initLogger()` and `GetLogger()`?

File: internal/services/recommender.go [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+		return
+	}
+	if err := validate.Struct(kafkaMsg); err != nil {
+		log.Errorf("Invalid kafka message: %s", err)
+		return
+	}
+
+	if err := processor.List_recommendations(kafkaMsg); err != nil {
+		log.Errorf("Unable to list recommendation for: %v", err)
+	}
```
Comment (upadhyeammit) [inline internal/services/recommender.go:34]: This whole block is same as `if..elseif..else`? or maybe you can use `switch` without any condition?

**Review by upadhyeammit — COMMENTED** [review]
I understand this is Draft, however I am trying to soak my eyes into the code. Request to check, thank you!

File: internal/services/recommender.go (hunk 2 above)
Comment (patilsuraj767) [reply internal/services/recommender.go:34]: I am not sure how we can use `switch` here, also shifting to `if..elseif..else` will make code messy. Can you add code suggestion.

File: internal/logging/logging.go (hunk 1 above)
Comment (patilsuraj767) [reply internal/logging/logging.go:31]: `GetLogger()` will help us to get the global instance of `log` which will be initialized once in the whole program execution. 
I am implementing the singleton design pattern here. 

In future we will be initializing AWS cloudwatch logging under `initLogger()`

Comment (upadhyeammit) [reply internal/logging/logging.go:31]: then makes sense!

File: internal/services/recommender.go (hunk 2 above)
Comment (upadhyeammit) [inline internal/services/recommender.go]: I still can be wrong, this is from my recent learning https://go.dev/tour/flowcontrol/11

```suggestion
	switch {
	case !json.Valid([]byte(msg.Value))
		log.Errorf("Received message on kafka topic is not vaild JSON: %s", msg.Value)
	case err := json.Unmarshal(msg.Value, &kafkaMsg); err != nil
		log.Errorf("Unable to decode kafka message: %s", msg.Value)
	case err := validate.Struct(kafkaMsg); err != nil
		log.Errorf("Invalid kafka message: %s", err)
	case err := processor.List_recommendations(kafkaMsg); err != nil
		log.Errorf("Unable to list recommendation for: %v", err)
	}
```

Comment (patilsuraj767) [reply internal/services/recommender.go]: I doubt this will work. Have you tested if the code flow is working as expected with these changes?

File: internal/model/rh_account.go [hunk 3]
```diff
@@ -1,7 +1,20 @@
 package model
 
+import (
+	database "github.com/redhatinsights/ros-ocp-backend/internal/db"
+)
+
 type RHAccount struct {
 	ID      uint   `gorm:"primaryKey;not null;autoIncrement"`
 	Account string `gorm:"type:text;unique"`
 	OrgId   string `gorm:"type:text;unique"`
```
Comment (saltgen) [inline internal/model/rh_account.go:10]: @patilsuraj767 As discussed, let's add a not null attribute for OrgID?

**Comment by saltgen** [comment]
@patilsuraj767 Let me know if this is good to merge

**Comment by patilsuraj767** [comment]
@saltgen We can merge this PR.

**Review by saltgen — APPROVED** [review]

**Comment by pgarciaq** [comment]
## Implementation Plan — GPU Utilization Radar Chart

### Design Decisions

1. **4-axis radar chart** with axes: SM Activity, Tensor Core Activity, DRAM Bandwidth, VRAM Usage
2. **Placement**: GPU breakdown pages only (MIG + Timeslicing), NOT container detail pages
3. **FB normalization**: `fb_usage_max_mib / total_fb_mib` to get a 0–1 fraction matching the other 3 metrics
4. **Library**: Raw Victory (`VictoryChart` polar + `VictoryPolarAxis` + `VictoryArea`) since PatternFly Charts doesn't wrap polar components
5. **Layout**: Radar chart rendered side-by-side with the existing VRAM gauge using PatternFly Grid
6. **Accessibility**: Hidden data table with actual percentages for screen readers (same pattern as VRAM gauge)

### Implementation Steps

- Fix TypeScript types: add `tensor_pipe_active_avg` and `dram_active_avg` to both `GPUMIGRecommendationData` and `GPUTimeslicingRecommendationData`
- Create `gpuUtilizationRadarChart.tsx` component with raw Victory polar chart
- Extend `GpuVisualInsightsSection` to accept utilization props and render radar chart alongside VRAM gauge
- Wire props through both `gpuMigBreakdown.tsx` and `gpuTimeslicingBreakdown.tsx`
- Add i18n messages for chart title, axis labels, and accessibility
- Write unit tests for the radar chart component
- Update CHANGELOG

Implementation in `koku-ui` repo.

**Comment by pgarciaq** [comment]
## Implementation Complete — GPU Utilization Radar Chart

### Files Changed (koku-ui repo)

| File | Change |
|------|--------|
| `apps/koku-ui-ros/src/api/ros/recommendations.ts` | Added `sm_active_avg`, `tensor_pipe_active_avg`, `dram_active_avg`, `fb_usage_max_mib`, `total_fb_mib` to `GPUMIGRecommendationData` and `GPUTimeslicingRecommendationData` |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuVisualInsights/gpuUtilizationRadarChart.tsx` | **New** — 4-axis polar radar chart (Victory `VictoryChart` + `VictoryArea` + `VictoryPolarAxis`) |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuVisualInsights/gpuUtilizationRadarChart.test.tsx` | **New** — 7 unit tests covering rendering, edge cases, accessibility |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuVisualInsights/gpuVisualInsightsSection.tsx` | Extended with new utilization props, PatternFly Grid layout (radar + VRAM gauge side-by-side) |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuVisualInsights/index.ts` | Added `GpuUtilizationRadarChart` export |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuMigBreakdown/gpuMigBreakdown.tsx` | Wired SM, Tensor, DRAM utilization props to `GpuVisualInsightsSection` |
| `apps/koku-ui-ros/src/routes/optimizations/optimizationsBreakdown/gpuTimeslicingBreakdown/gpuTimeslicingBreakdown.tsx` | Wired utilization props; removed `as any` casts (types now include those fields) |
| `apps/koku-ui-ros/src/locales/messages.ts` | Added 8 i18n messages for chart title, axis labels, accessibility |
| `CHANGELOG.md` | Added entry for GPU Subsystem Utilization radar chart |

### Tests

All 7 new tests pass. All 8 existing VRAM gauge tests pass. No new TypeScript errors introduced.
